CONTRIBUTING.md

```diff
@@ -0,0 +1,23 @@
+# Contributing
+
+If you want to contribute to the project, you shall fork it first and make your changes starting from your own fork.
+
+## Making changes
+
+Whenever new changes are made, make sure that these are grouped together under the same branch on your fork. This will ensure that whenever you are working on different unrelated tasks, the work you want to merge to master won't be affected.
+
+Every commit must:
+- contain changes only specific to a single task, which when too big, must split into subtasks. Tasks can be organised and tracked using the GitHub issues tab;
+- every commit must hold a meaningful and concise summary of the changes that the commit brings. For example, if you are creating a new function called `calculate_formula`, this should be of the form of: `Add calculate_formula function`. Make sure to use only *ONE* verb. If multiple verbs are needed to describe your commit, this means that your commit must be split into multiple commits! (i.e. it's too big)
+- an in-depth description of the commit may also be added if needed to understand the scope of the commit;
+- if any commit is related to an issue or multiple issues, you can reference these verbally in your commit description.
+
+## Pushing the changes
+
+When your patch, a collection of interdependent commits, is ready, you may create a pull request to master. Every pull request must be reviewed by another member for approval to merge. If CI testing is employed, a pass is also required for the approval.
+
+### Reverting the changes
+
+If any of your commits in your pending PR have comments that require attention, you must amend your changes to keep a clean history. For example, if your commit `Add calculate_formula function`, meaning that a new function is being added, is lacking a proper documentation, you must amend it instead of creating a new one. This will ensure that your function will come together as a whole into one single commit, instead of being split into multiple commits that are about the same thing resulting in a decline of history consistency.  
+
+nioadfoudisfusdjhfdshgfioudj
```

Comment:
Oops!

Reply:
Also the commit name is not fine! Change with something like: Add CONTRIBUTING.md